Sync/Synchronizers/LoginsSynchronizer.swift

```diff
@@ -107,20 +108,22 @@ public class LoginsSynchronizer: IndependentRecordSynchronizer, Synchronizer {
         }
     }
 
-    private func uploadModifiedLogins(logins: [Login], lastTimestamp: Timestamp, fromStorage storage: SyncableLogins, withServer storageClient: Sync15CollectionClient<LoginPayload>) -> DeferredTimestamp {
-        return self.uploadRecords(logins.map(makeLoginRecord), by: 50, lastTimestamp: lastTimestamp, storageClient: storageClient) {
-            storage.markAsSynchronized($0.success, modified: $0.modified)
-        }
-    }
+    private func uploadChangedRecords<T>(deleted: [GUID], modified: [GUID], records: [Record<T>], lastTimestamp: Timestamp,
+                                      storage: SyncableLogins, withServer storageClient: Sync15CollectionClient<T>) -> Success {
+        let onUpload: (POSTResult) -> DeferredTimestamp = { result in
+            let uploaded = result.success
 
-    private func uploadDeletedLogins(guids: [GUID], lastTimestamp: Timestamp, fromStorage storage: SyncableLogins, withServer storageClient: Sync15CollectionClient<LoginPayload>) -> DeferredTimestamp {
+            // Figure out which ones the server deleted and which ones were modified
+            let guidsDeleted = deleted.filter(uploaded.contains)
+            let guidsModified = modified.filter(uploaded.contains)
```

Comment:
Consider using sets here; set intersection should be faster, and we don't care about ordering.